src/EzPlatformCoreBundle/bundle/DependencyInjection/EzPlatformCoreExtension.php

```diff
@@ -47,4 +51,255 @@ public function prepend(ContainerBuilder $container): void
+                    continue;
+                }
+
+                $loader = new Loader\YamlFileLoader($container, new FileLocator($projectDir . '/config/packages/cache_pool'));
+                $loader->load('cache.redis.yaml');
+
+                $container->setParameter('cache_pool', 'cache.redis');
+                $container->setParameter('cache_dsn', sprintf('%s:%d', $endpoint['host'], $endpoint['port']) . '?retry_interval=3');
+            }
+        } elseif (isset($relationships['cache'])) {
+            // Fallback to memcached if here (deprecated, we will only handle redis here in the future)
+            foreach ($relationships['cache'] as $endpoint) {
+                if ($endpoint['scheme'] !== 'memcached') {
+                    continue;
+                }
+
+                @trigger_error('Usage of Memcached is deprecated, redis is recommended', E_USER_DEPRECATED);
+
+                $container->setParameter('cache_pool', 'cache.memcached');
+                $container->setParameter('cache_dsn', sprintf('%s:%d', $endpoint['host'], $endpoint['port']));
+
+                $loader = new Loader\YamlFileLoader($container, new FileLocator($projectDir . '/config/packages/cache_pool'));
+                $loader->load('cache.memcached.yaml');
+            }
+        }
+
+        // Use Redis-based sessions if possible. If a separate Redis instance
+        // is available, use that.  If not, share a Redis instance with the
+        // Cache.  (That should be safe to do except on especially high-traffic sites.)
+        if (isset($relationships['redissession'])) {
+            foreach ($relationships['redissession'] as $endpoint) {
+                if ($endpoint['scheme'] !== 'redis') {
+                    continue;
+                }
+
+                $container->setParameter('ezplatform.session.handler_id', 'ezplatform.core.session.handler.native_redis');
+                $container->setParameter('ezplatform.session.save_path', sprintf('%s:%d', $endpoint['host'], $endpoint['port']));
+            }
+        } elseif (isset($relationships['rediscache'])) {
+            foreach ($relationships['rediscache'] as $endpoint) {
+                if ($endpoint['scheme'] !== 'redis') {
+                    continue;
+                }
+
+                $container->setParameter('ezplatform.session.handler_id', 'ezplatform.core.session.handler.native_redis');
+                $container->setParameter('ezplatform.session.save_path', sprintf('%s:%d', $endpoint['host'], $endpoint['port']));
+            }
+        }
+
+        if (isset($relationships['solr'])) {
+            foreach ($relationships['solr'] as $endpoint) {
+                if ($endpoint['scheme'] !== 'solr') {
+                    continue;
+                }
+
+                $container->setParameter('search_engine', 'solr');
+
+                $container->setParameter('solr_dsn', sprintf('http://%s:%d/%s', $endpoint['host'], $endpoint['port'], 'solr'));
```

Comment:
As you might remember from previous week, this parameter is slightly different in Commerce due to Solarium coupling config conflict. Is there a plan to overcome this issue? My idea was to unify configuration so the version from Commerce works for OSS & DXP, but haven't investigated it yet.

Reply:
Yes this is why a few lines below there is `if` statement handling ezcommerce case: https://github.com/ezsystems/ezplatform-core/pull/27/files/b8ce23f032076d82bb56a049830631e0c44e0d97#diff-f72b922417794d0e5c2291bf6eddcfdf0e1e32a18d00f5ff3e8b46c514baa100R236-R244

Reply:
> Yes this is why a few lines below there is `if` statement handling ezcommerce case: https://github.com/ezsystems/ezplatform-core/pull/27/files/b8ce23f032076d82bb56a049830631e0c44e0d97#diff-f72b922417794d0e5c2291bf6eddcfdf0e1e32a18d00f5ff3e8b46c514baa100R236-R244

Uh, difficult to spot in this spaghetti. OK.